src/libraries/System.Text.Json/gen/JsonSourceGenerator.DiagnosticDescriptors.cs

```diff
@@ -88,6 +88,14 @@ internal static class DiagnosticDescriptors
                 category: JsonConstants.SystemTextJsonSourceGenerationName,
                 defaultSeverity: DiagnosticSeverity.Warning,
                 isEnabledByDefault: true);
+
+            public static DiagnosticDescriptor JsonUnsupportedLanguageVersion { get; } = new DiagnosticDescriptor(
+                id: "SYSLIB1042",
```

Comment:
Update `list-of-diagnostics.md`?

Reply:
Good call, I'll follow-up with another PR since other recently added diagnostics are not documented either.

Reply:
https://github.com/dotnet/runtime/pull/87980